### ISSUE TITLE
Google account login page doesn't work

### ISSUE THREAD

**Issue by eixan** [omarroth/invidious#292]
I tried it on mobile brave 1.0.72 chromium 71.0.3578.99, mobile firefox 64.0.2, and mobile chrome 71.0.3578.99

Running android oreo kernel version 3.18.71 build number:R16NW

It says incorrect password

**Comment by omarroth** [comment]
Would you mind sending me an email at omarroth@protonmail.com? I'd like to set up, or help you set up, a private instance to help debug this.

**Comment by sdtqwe4ty** [comment]
> Would you mind sending me an email at [omarroth@protonmail.com](mailto:omarroth@protonmail.com)? I'd like to set up, or help you set up, a private instance to debug this.

Hi I can't access my account:mathcreative@gmail.com. So if you posted a followup to my last reply then I missed it.

**Comment by omarroth** [comment]
I pushed out 699f85e773c7e69ac081c2361be3de89541accf5 which has fixed login for several people. If you are still having trouble I'd recommend getting in contact with me privately using the email I mentioned above.

**Comment by idnovic** [comment]
I can not login. I did use an app specific password.

**Comment by omarroth** [comment]
Currently app-specific passwords are not supported. I have not found any information on how they would be implemented, and I don't have any way to test them. See also rg3/youtube-dl#1336.

From the limited information I can gather, for example [this](https://duo.com/blog/bypassing-googles-two-factor-authentication) article from 2013, authentication using application-specific passwords requires OAuth credentials or some other form of registration with Google, which is undesirable for this project.

**Comment by idnovic** [comment]
good to know. I did try normal login with 2fa but I got an error "2fa seems to be disabled" or something like that.

**Comment by omarroth** [comment]
Currently only Authenticator (or similar apps) and text message are supported, so that may be the problem you're encountering.

**Comment by idnovic** [comment]
by authenticator you mean totp right?

**Comment by omarroth** [comment]
Correct, so Authenticator, Authy, and codes from similar apps should work.

**Comment by idnovic** [comment]
It did not for me. Let me try again

**Comment by idnovic** [comment]
Login failed. This may be because two-factor authentication is not enabled on your account.

**Comment by ewdissi** [comment]
Hi, ive now the same Issue. Trying to login via google throws the error: " Login failed. This may be because two-factor authentication is not enabled on your account. ". But 2factor is enable. I'am using "andOTP" to generating my codes. Any ideas?

**Comment by idnovic** [comment]
well it does not work and google security page tells me than their was a login to my account from usa. I live in europe.

**Comment by omarroth** [comment]
Apologies for not responding sooner. I've pushed a couple changes with f1d7aa09e45a16216a8a71789de11c520ad72ab5. I'd suggest trying to login again to see if the mentioned changes fixed the issue.

> google security page tells me than their was a login to my account from usa

This is because invidio.us is hosted in NJ, USA. See [here](https://www.reddit.com/r/privacytoolsIO/comments/9fwp3f/is_there_good_place_like_hooktube/e6c9foe/) for more info.

**Comment by ewdissi** [comment]
> 
> 
> Apologies for not responding sooner. I've pushed a couple changes with [f1d7aa0](https://github.com/omarroth/invidious/commit/f1d7aa09e45a16216a8a71789de11c520ad72ab5). I'd suggest trying to login again to see if the mentioned changes fixed the issue.
> 
> > google security page tells me than their was a login to my account from usa
> 
> This is because invidio.us is hosted in NJ, USA. See [here](https://www.reddit.com/r/privacytoolsIO/comments/9fwp3f/is_there_good_place_like_hooktube/e6c9foe/) for more info.

Didn't work for me. Now the login directly broke up. Even not the 2fa field where shown after the login

**Comment by n3m1s1s94** [comment]
Hi, I try logging in with google account but it does not work for me. I understand after looking for solutions on your page that I can import my subscriptions and such but I'd still prefer to log in to my google account to retain options to comments and accessing my playlists etc. 
I get the login failed because 2 factor authentication is turned off message each time though I do have it turned on and set to send me an SMS code. I really like and appreciate the work you have made and really want to use invidio as my main goto URL for my youtube needs instead of the original site seeing as how lightweight and adfree invidio is and also it works so much better with qute but this is preventing me from doing so and I have been trying for almost 2 weeks now. I was wondering if there was anything you had in mind I could try. I am not a computer person by proffesion so please treat me as a dummy if there is something technincal involved. Thank you. 
PS -  I really think this is amazing. Does invidio have a patreon or kickstarter or something btw ?

**Comment by omarroth** [comment]
Would you mind contacting me privately at omarroth@protonmail.com ? You might also take a look at [this comment](https://github.com/omarroth/invidious/issues/154#issuecomment-498887873) in #154.

**Comment by FuturePilot** [comment]
I am also unable to login with Google. The error says login failed because 2FA is not enabled. I definitely have 2FA enabled. From the other comment, this is what I get with youtube-dl

  $ youtube-dl -u $USERNAME -v https://www.youtube.com/feed/history
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-u', u'PRIVATE', u'-v', u'https://www.youtube.com/feed/history']
Type account password and press [Return]: 
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.06.08
[debug] Python version 2.7.16 (CPython) - Linux-5.0.0-16-generic-x86_64-with-Ubuntu-19.04-disco
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3
[debug] Proxy map: {}
[youtube:history] Downloading login page
[youtube:history] Looking up account info
[youtube:history] Logging in
WARNING: Unable to log in: HTTP Error 400: Bad Request
[youtube:history] Youtube History: Downloading webpage
[download] Downloading playlist: Youtube History
[youtube:history] playlist Youtube History: Downloading 0 videos
[download] Finished downloading playlist: Youtube History

**Comment by omarroth** [comment]
Thanks! Mind also providing what form of authentication you're using?

**Comment by omarroth** [comment]
Just pushed a couple changes with fa6bf21cd15f8f8e7c2f3bb66744402eb99dd3bc.

If login fails it will now provide a traceback with some more info, example:
```
Expected Array for #[](index : Int), not Int64
Traceback:
  Getting lookup...done, returned 200.
  Getting challenge...done, returned 400.
```

**Comment by xArgv** [comment]
Hey mate I get this error when I try to login with google account

Couldn't get SID.
Traceback:
Getting lookup...done, returned 200.
Getting challenge...done, returned 200.
User has 2FA.
Using challenge 9.
Logging in...

I don't have 2fa but I have phone number linked to the account and google sometimes asks for sms verification?

**Comment by FuturePilot** [comment]
@omarroth I'm just using my Google password. Should we be using an app specific password?

**Comment by omarroth** [comment]
@xArgv: Thanks. I can reproduce this locally. It appears to be a page presented after login that Invidious chokes on. Should be a mostly straight-forward fix when I have a free moment.

@FuturePilot: No, your regular password should be fine. Would you mind attempting to login again and providing any error messages you encounter?

**Comment by FuturePilot** [comment]
@omarroth I was able to successfully login this time. I was presented with a captcha after entering my password.

**Comment by omarroth** [comment]
If you encountered a captcha then you probably made an Invidious account rather than linking with your Google account. You should be able to delete that account, then login using “Login with Google”.

**Comment by FuturePilot** [comment]
Woops! I wasn't paying attention to what I was doing. Tried with my Google account and I was prompted for my 2FA code. The first time I got a 504 gateway timeout error but the second time it succeeded.

**Comment by omarroth** [comment]
Ha, no worries. The 504 is likely caused by importing a large number of subscriptions, which should hopefully be fixed soon.

@xArgv Pushed a3164177f8e7e3d323006e08755230f850a5fdd9 which should fix the issue you're encountering.
